pkg/services/publicdashboards/api/api.go

```diff
@@ -35,21 +38,27 @@ func ProvideApi(
 	ac accesscontrol.AccessControl,
 	features *featuremgmt.FeatureManager,
 	md publicdashboards.Middleware,
+	cfg *setting.Cfg,
 ) *Api {
 	api := &Api{
 		PublicDashboardService: pd,
-		RouteRegister:          rr,
-		AccessControl:          ac,
-		Features:               features,
-		Log:                    log.New("publicdashboards.api"),
 		Middleware:             md,
+		accessControl:          ac,
+		cfg:                    cfg,
+		features:               features,
+		log:                    log.New("publicdashboards.api"),
+		routeRegister:          rr,
 	}
 
-	// attach api if PublicDashboards feature flag is enabled
-	if features.IsEnabledGlobally(featuremgmt.FlagPublicDashboards) {
+	// register endpoints if the feature is enabled
+	if features.IsEnabledGlobally(featuremgmt.FlagPublicDashboards) && cfg.PublicDashboardsEnabled {
 		api.RegisterAPIEndpoints()
 	}
 
+	if !features.IsEnabledGlobally(featuremgmt.FlagPublicDashboards) {
+		api.log.Warn("[Deprecated] The publicDashboards feature toggle will be removed in Grafana v11. If you want to disable the public dashboards feature, you should use the public_dashboards.enabled setting.")
```

Comment:
This might be a bit hidden but I didn't find another place to show this warning.

Reply:
What about server start-up? Do we have a convention about where to put these warnings?

Reply:
This will already be at server start-up but this will be in a middle of all services start-up logs. I don't think we have a convention about this. The best would have been in the UI but we can't really do that here as it's only when the feature is disabled and when it is, there is no UI.

Reply:
I think this place is perfect, the UI should be useful only if we want to communicate something to users about a feature but this is for the Grafana instance admins.